src/Symfony/Component/Routing/Loader/LoaderHelper.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Routing\Loader;
+
+/**
+ * Common parsing logic between route loaders.
+ *
+ * @author Samuel Roze <samuel.roze@gmail.com>
+ */
+trait LoaderHelper
+{
+    private function redirectToDefaults(string $redirect, bool $permanent = false): array
+    {
+        if ($this->looksLikeUrl($redirect)) {
+            $defaults['_controller'] = 'Symfony\Bundle\FrameworkBundle\Controller\RedirectController::urlRedirectAction';
+            $defaults['path'] = $redirect;
+        } else {
+            $defaults['_controller'] = 'Symfony\Bundle\FrameworkBundle\Controller\RedirectController::redirectAction';
```

Comment:
But there is a bug glitch here: this is in the `Routing` namespace, and it references something in `FrameworkBundle`.
We have to fix that.

Reply:
Good point. For the records, here's [@javiereguiluz's answer](https://github.com/symfony/symfony/issues/24640#issuecomment-338165615) when @stof's said the same thing. I need to think a bit about it.

Reply:
I see a few options:
1. A simple `class_exists` check on the `RedirectController` to have a user-friendly exception. This references the FrameworkBundle, but is not a hard coupling.
2. We use a `_redirect` attribute, that will be handled by a `ControllerResolverInterface` implementation within the FrameworkBundle.
3. We use a `_redirect` attribute that will be handled by a listener on `kernel.request` within the `HttpKernel` component